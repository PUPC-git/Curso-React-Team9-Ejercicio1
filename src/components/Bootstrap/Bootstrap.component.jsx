import React, { useState } from 'react';
import { useDispatch } from 'react-redux/es/exports';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Link } from 'react-router-dom';
import { faMoon, faSun } from '@fortawesome/free-regular-svg-icons';
import { cambioStyle } from '../../redux/actions/actions';
import './Bootstrap.component.scss';

export default function Bootstrap() {
    const [esDeDia, setDia] = useState(true);
    const dispatch = useDispatch();
  
    let texto = esDeDia ?
      (<div><FontAwesomeIcon icon={faMoon}/></div>) :
      (<div><FontAwesomeIcon icon={faSun}/></div>);
      
    let cambio = () => {
        setDia(!esDeDia);
        localStorage.setItem('cambioStyle', !esDeDia);
        console.log(esDeDia);
        dispatch(cambioStyle(esDeDia));
    }
    return (
      /* Contenido JSX de la p√°gina */
      
       <div className="container vh-10 mw-100">
     
         <button className="btn btn-primary" onClick={() => cambio()}>
           {texto}
         </button>
     
     </div>
    );
  }