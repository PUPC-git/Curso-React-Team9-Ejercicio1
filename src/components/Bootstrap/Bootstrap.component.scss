/* Estilos propios de la p√°gina Bootstrap */

