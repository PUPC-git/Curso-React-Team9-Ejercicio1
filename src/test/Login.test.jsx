import { create } from "react-test-renderer";
import {render, screen} from '@testing-library/react';
import LoginButton from "../components/LoginButton.component";
import LogoutButton from "../components/LogoutButton.component";
import Bootstrap from "../components/Bootstrap/Bootstrap.component";
import { unmountComponentAtNode } from "react-dom";
//import RegisterLogin from "../components/RegisterLogin/RegisterLogin.component";

let props = {
    onClick: 'Prueba'
}
	
let container = null;

beforeEach(() => {
    // configurar un elemento del DOM como objetivo del renderizado
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  
  afterEach(() => {
    // limpieza al salir
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});


describe('test del login', () => {
    test("LoginButton test", () => {
        const button = create(LoginButton(props));
        expect(button).toMatchSnapshot('');      
    });

    test("LogoutButton test", () => {
        const button = create(LogoutButton(props));
        expect(button).toMatchSnapshot('');      
    });

    
    test("Boostrap test", () => {
        render(<Bootstrap/>)
        expect(screen.getByRole('button')).toBeInTheDocument();
    });
    
    /*  --Da error - Cannot use import statement outside a module (useDispatch linea 5) 
    test("Mensajes Header test", () => {
        render(<MensajesHeader/>)
        expect(screen.getByRole('table')).toBeInTheDocument();
    });*/

    /*  Da error - Cannot use import statement outside a module  (useDispatch linea 5)
    test("Mensajes Table test", () => {
        render(<MensajesTable/>)
        expect(screen.getByRole('table')).toBeInTheDocument();
    });*/

    /* Da error - useNavigate() may be used only in the context of a <Router> component
    test("Navigation test", () => {
        render(<RegisterLogin/>)
        expect(screen.getByRole('nav')).toBeInTheDocument();
    });*/

    /*
    it("Mock RegisterLogin test", async () => {
        jest.mock('./../components/RegisterLogin/RegisterLogin', () => {
            (<div>ffdf</div>)
        }) 
        
    });
*/
});