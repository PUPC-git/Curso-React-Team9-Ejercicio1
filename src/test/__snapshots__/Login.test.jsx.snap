// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test del login LoginButton test 1`] = `
<button
  class="btn btn-primary"
  onClick="Prueba"
>
  Conectarse
</button>
`;

exports[`test del login LogoutButton test 1`] = `
<button
  class="btn btn-primary"
  onClick="Prueba"
>
  Desconectarse
</button>
`;
